docs/aws/audit/elbmonitoring/rules/elb_secure_listener_updated_tls.mdx

```diff
@@ -24,6 +24,232 @@ SOC2, AWSWAF
+
+   ```sh
+   aws elb create-load-balancer-listeners --load-balancer-name my-load-balancer --listeners "Protocol=HTTPS,LoadBalancerPort=443,InstanceProtocol=HTTPS,InstancePort=443,SSLCertificateId=arn:aws:iam::123456789012:server-certificate/my-server-cert"
+   ```
+
+2. **Delete Insecure Listeners:**
+   Remove any existing listeners that use insecure protocols such as HTTP. Use the `delete-load-balancer-listeners` command to remove these listeners.
+
+   ```sh
+   aws elb delete-load-balancer-listeners --load-balancer-name my-load-balancer --load-balancer-ports 80
+   ```
+
+3. **Verify Listener Configuration:**
+   Confirm that your ELB is configured with only secure listeners by describing the load balancer and checking the listener details.
+
+   ```sh
+   aws elb describe-load-balancers --load-balancer-names my-load-balancer
+   ```
+
+4. **Enforce Security Policies:**
+   Apply a security policy that enforces the use of secure protocols and ciphers. Use the `set-load-balancer-policies-of-listener` command to set the security policy.
+
+   ```sh
+   aws elb set-load-balancer-policies-of-listener --load-balancer-name my-load-balancer --load-balancer-port 443 --policy-names ELBSecurityPolicy-2016-08
+   ```
+
+By following these steps, you can ensure that your ELBs are configured to use secure listeners only, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that Elastic Load Balancers (ELBs) in AWS use secure listeners only, you can use the AWS SDK for Python, known as Boto3. Below are the steps to prevent this misconfiguration using Python scripts:
+
+### 1. Install Boto3
+First, ensure you have Boto3 installed. You can install it using pip if you haven't already:
+```bash
+pip install boto3
+```
+
+### 2. Create a Boto3 Session
+Set up a Boto3 session to interact with AWS services. Make sure you have the necessary AWS credentials configured.
+```python
+import boto3
+
+# Create a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+# Create an ELB client
+elb_client = session.client('elb')
+```
+
+### 3. List All Load Balancers
+Retrieve a list of all load balancers to check their listener configurations.
+```python
+# List all load balancers
+load_balancers = elb_client.describe_load_balancers()
+
+for lb in load_balancers['LoadBalancerDescriptions']:
+    lb_name = lb['LoadBalancerName']
+    print(f"Checking listeners for load balancer: {lb_name}")
+```
+
+### 4. Check and Update Listeners to Use Secure Protocols
+Check the listeners for each load balancer and update them to use secure protocols (HTTPS or SSL) if they are not already configured.
+```python
+# Iterate through each load balancer
+for lb in load_balancers['LoadBalancerDescriptions']:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>